sound/soc/sof/pcm.c

```diff
@@ -361,6 +365,12 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_START;
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
+		if (spcm->stream[substream->stream].d0i3_compatible) {
+			/* KWD keep working, don't handle DAPM PMD event */
+			sdev->kwd_at_s0i3 = true;
```

Comment:
no, this is horrible. Don't introduce something that is so specific in a generic code. NAK on this one.